pavelib/paver_tests/test_safelint.py

```diff
@@ -0,0 +1,50 @@
+"""
+Tests for paver quality tasks
+"""
+import unittest
+from mock import patch
+
+import pavelib.quality
+from paver.easy import BuildFailure
+
+
+class TestPaverSafeLint(unittest.TestCase):
+    """
+    For testing run_safelint
+    """
+
+    def setUp(self):
+        super(TestPaverSafeLint, self).setUp()
+
+        # Mock the paver @needs decorator
+        self._mock_paver_needs = patch.object(pavelib.quality.run_safelint, 'needs').start()
+        self._mock_paver_needs.return_value = 0
+
+        # Mock shell commands
+        patcher = patch('pavelib.quality.sh')
+        self._mock_paver_sh = patcher.start()
+
+        # Cleanup mocks
+        self.addCleanup(patcher.stop)
+        self.addCleanup(self._mock_paver_needs.stop)
+
+    @patch.object(pavelib.quality, '_write_metric')
+    @patch.object(pavelib.quality, '_prepare_report_dir')
+    @patch.object(pavelib.quality, '_get_count_from_last_line')
+    def test_safelint_violation_number_not_found(self, mock_count, mock_report_dir, mock_write_metric):  # pylint: disable=unused-argument
```

Comment:
NIT: If you prepend the unused argument names with "_" (or anything matching the configured [dummy-variables-rgx](https://github.com/edx/edx-platform/blob/master/pylintrc#L154)) then you don't need to disable the warning.